src/csharp/Grpc.Core.Api/Status.cs

```diff
@@ -31,48 +33,61 @@ public struct Status
         /// </summary>
         public static readonly Status DefaultCancelled = new Status(StatusCode.Cancelled, "");
 
-        readonly StatusCode statusCode;
-        readonly string detail;
-
         /// <summary>
         /// Creates a new instance of <c>Status</c>.
         /// </summary>
         /// <param name="statusCode">Status code.</param>
         /// <param name="detail">Detail.</param>
-        public Status(StatusCode statusCode, string detail)
+        public Status(StatusCode statusCode, string detail) : this(statusCode, detail, null)
         {
-            this.statusCode = statusCode;
-            this.detail = detail;
         }
 
         /// <summary>
-        /// Gets the gRPC status code. OK indicates success, all other values indicate an error.
+        /// Creates a new instance of <c>Status</c>.
+        /// Users should not use this constructor, except for creating instances for testing.
+        /// The debug error string should only be populated by gRPC internals.
         /// </summary>
-        public StatusCode StatusCode
+        /// <param name="statusCode">Status code.</param>
+        /// <param name="detail">Detail.</param>
+        /// <param name="debugErrorException">Optional internal error details.</param>
+        public Status(StatusCode statusCode, string detail, Exception debugErrorException)
         {
-            get
-            {
-                return statusCode;
-            }
+            StatusCode = statusCode;
+            Detail = detail;
+            DebugErrorException = debugErrorException;
         }
 
+        /// <summary>
+        /// Gets the gRPC status code. OK indicates success, all other values indicate an error.
+        /// </summary>
+        public StatusCode StatusCode { get; }
+
         /// <summary>
         /// Gets the detail.
         /// </summary>
-        public string Detail
-        {
-            get
-            {
-                return detail;
-            }
-        }
+        public string Detail { get; }
+
+        /// <summary>
+        /// In case of an error, this field may contain additional error details to help with debugging.
+        /// This field will be only populated on a client and its value is generated locally,
+        /// based on the internal state of the gRPC client stack (i.e. the value is never sent over the wire).
+        /// Note that this field is available only for debugging purposes, the application logic should
+        /// never rely on values of this field (it should use <c>StatusCode</c> and <c>Detail</c> instead).
+        /// Example: when a client fails to connect to a server, this field may provide additional details
+        /// why the connection to the server has failed.
+        /// </summary>
+        public Exception DebugErrorException { get; }
```

Comment:
Experimental API?

Reply:
DebugErrorException -> DebugException? Exception kind of already indicates that it is from an error.

Reply:
Done.